robotidy/transformers/aligners_core.py

```diff
@@ -205,57 +211,89 @@ def visit_KeywordCall(self, node):  # noqa
             return node
         if self.skip.keyword_call(node):
             return node
+        return self.align_node(node, check_length=self.split_too_long)
+
+    @skip_if_disabled
+    def visit_Tags(self, node):  # noqa
+        if node.errors:
+            return node
+        return self.align_node(node, check_length=False)
+
+    def align_node(self, node, check_length):
         lines = list(tokens_by_lines(node))
         indent = Token(Token.SEPARATOR, self.indent * self.formatting_config.indent)
-        separator = self.formatting_config.space_count
-        aligned_statement = []
+        aligned_lines = []
         for line in lines:
-            prev_overflow_len = 0
-            fixed_separator = False
-            aligned_statement.append(indent)
-            if is_blank_multiline(line):  # ...\n edge case
-                line[-1].value = line[-1].value.lstrip(" \t")  # normalize eol from '  \n' to '\n'
-                aligned_statement.extend(line)
-                continue
-            tokens, comments = separate_comments(line)
-            if len(tokens) < 2:  # only happens with weird encoding, better to skip
+            aligned_line = self.align_line(line, indent)
+            if aligned_line is None:
                 return node
-            column = 0
-            for token in tokens[:-2]:
-                aligned_statement.append(token)
-                if fixed_separator or (self.skip.return_values and token.type == Token.ASSIGN):
-                    aligned_statement.append(Token(Token.SEPARATOR, separator * " "))
-                    continue
-                width = self.get_width(column)
-                if width == 0:
-                    separator_len = round_to_four(len(token.value) + separator) - len(token.value)
+            if check_length and self.is_line_too_long(aligned_line):
+                split_node = self.split_too_long_node(node)
+                return self.align_node(split_node, check_length=False)
+            aligned_lines.extend(aligned_line)
+        return Statement.from_tokens(aligned_lines)
+
+    visit_Arguments = visit_Setup = visit_Teardown = visit_Timeout = visit_Template = visit_Return = visit_Tags
+
+    def align_line(self, line, indent):
```

Comment:
Maybe some docstring here with explanation for such a long ifology? :D

Reply:
Yeah, I plan to - I already reduced the code a bit (by moving part of the lines outside). I will try to add some docs and move to separate methods whenever feasible.